app/views/topics/_form.html.erb

```diff
@@ -37,9 +37,10 @@
       <div class="well">
         <h3>Grouping</h3>
         <ul class="list-group topic-section-list js-topic-section-list">
-          <%= f.fields_for :topic_sections do |topic_section_form| %>
+          <%= f.fields_for :topic_sections, f.object.topic_sections.sort_by(&:position) do |topic_section_form| %>
```

Comment:
There is probably a N+1 to be squashed here with an `.includes(:guides)`. It may well be needed in the controller too where we build new topic sections.

The world will keep turning without fixing it off course.. Performance probably isn't a big issue anywhere in this app :)